src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -0,0 +1,602 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Threading;
+using System.Runtime.CompilerServices;
+using System.Runtime.Versioning;
+using Internal.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Internal enumeration used by the runtime to indicate the scenario for which ComWrappers is being used.
+    /// </summary>
+    internal enum ComWrappersScenario
+    {
+        Instance = 0,
+        TrackerSupportGlobalInstance = 1,
+        MarshallingGlobalInstance = 2,
+    }
+
+    /// <summary>
+    /// Class for managing wrappers of COM IUnknown types.
+    /// </summary>
+    public abstract partial class ComWrappers
+    {
+        const int DispatchAlignmentThisPtr = 16;
+        private static readonly ConditionalWeakTable<object, ManagedObjectWrapperHolder> CCWTable = new ConditionalWeakTable<object, ManagedObjectWrapperHolder>();
+
+        /// <summary>
+        /// ABI for function dispatch of a COM interface.
+        /// </summary>
+        public partial struct ComInterfaceDispatch
+        {
+            /// <summary>
+            /// Given a <see cref="System.IntPtr"/> from a generated Vtable, convert to the target type.
+            /// </summary>
+            /// <typeparam name="T">Desired type.</typeparam>
+            /// <param name="dispatchPtr">Pointer supplied to Vtable function entry.</param>
+            /// <returns>Instance of type associated with dispatched function call.</returns>
+            public static unsafe T GetInstance<T>(ComInterfaceDispatch* dispatchPtr) where T : class
+            {
+                // See the dispatch section in the runtime for details on the masking below.
+                const long DispatchThisPtrMask = ~0xfL;
+                var comInstance = *(ComInterfaceInstance**)(((long)dispatchPtr) & DispatchThisPtrMask);
```

Comment:
That would be problematic, since & operator does not operates on the `nuint` and as such I still have to use widest type available to trim lowest bytes, and only then convert to nuint.

Reply:
I am not sure what you mean. `&` operator should work just fine on `nuint`. For example, `static nuint f(nuint x, nuint y) => x & y;` compiles fine for me.